### DISCUSSION TITLE
Overlay two labels into a dual channel label

### DISCUSSION THREAD

**Discussion by wxr521314** [Project-MONAI/MONAI#7801] [General]
I have data.nii.gzl. Label.nii.gz and label_2.nii.gz, I would like to add label.nii.gz, Label_2.nii.gz is superimposed into a two channel data. Can LoadNiftid be automatically completed? What should I do?

**Comment by wxr521314** [comment]
I have a dual channel 128 * 128 * 128 data, and I use EnsureChannelFirst(), torch.Size ([1, 2, 128, 128, 128, 1]), why is there an extra dimension? It shouldn't be torch.Size ([1, 2, 128, 128, 128,]) correct?

**Comment by KumoLiu** [comment]
> I have data.nii.gzl. Label.nii.gz and label_2.nii.gz, I would like to add label.nii.gz, Label_2.nii.gz is superimposed into a two channel data. Can LoadNiftid be automatically completed? What should I do?

You can use `ConcatItemsd` to help you concate.
https://github.com/Project-MONAI/MONAI/blob/e5afa4341b18149ca8d44f680f669b33ab423722/monai/transforms/utility/dictionary.py#L927

> I have a dual channel 128 * 128 * 128 data, and I use EnsureChannelFirst(), torch.Size ([1, 2, 128, 128, 128, 1]), why is there an extra dimension? It shouldn't be torch.Size ([1, 2, 128, 128, 128,]) correct?

`EnsureChannelFirst` is used to ensure the channel dim first, if you already have 2 as your channel dim, no need to add this transform.

Thanks.
(Move to discussion as more like a usage issue.)

**Comment by wxr521314** [comment]
If I don't use EnsureChannelFirst(),torch.Size is ([1, 128, 128, 128, 1, 2]). Is that right? Using nibabel to read a multi-channel nii.gz, does the result automatically add one more dimension?
